### PR TITLE
[WIP] Better template tests

### PR DESCRIPTION
**Pull request by Dignifiedquire**
As discussed in #312 this is the beginning improving the template tests using an html parser instead of regex assertions. 

cc @milmazz @josevalim

### CODE REVIEW

File: test/ex_doc/formatter/html/templates_test.exs [hunk 1]
```diff
@@ -83,19 +110,23 @@ defmodule ExDoc.Formatter.HTML.TemplatesTest do
     assert content =~ ~r{.*\"id\":\s*\"CompiledWithDocs\".*\"functions\".*\"example/2\".*}ms
     assert content =~ ~r{.*\"id\":\s*\"CompiledWithDocs\".*\"macros\".*\"example_1/0\".*}ms
     assert content =~ ~r{.*\"id\":\s*\"CompiledWithDocs\".*\"functions\".*\"example_without_docs/0\".*}ms
-    assert content =~ ~r{.*"CompiledWithDocs.Nested\"}ms
+    assert content =~ ~r{.*"CompiledWithDocs.Nested\"}ms # fix emacs highlighting "
```
Comment (milmazz) [inline test/ex_doc/formatter/html/templates_test.exs]: Is this just a note for you?

File: test/ex_doc/formatter/html/templates_test.exs [hunk 2]
```diff
@@ -83,19 +110,23 @@ defmodule ExDoc.Formatter.HTML.TemplatesTest do
...
     refute content =~ ~r{.*\"exceptions\":}ms
     refute content =~ ~r{.*\"protocols\":}ms
   end
 
   test "listing page has README link if present" do
     content = Templates.sidebar_template(doc_config, [], [], [], true)
-    assert content =~ ~r{<a href="README.html">README</a>}
+    links = Floki.find(content, ".sidebar-mainNav li")
+
+    assert_text Enum.at(links, 0), "README"
```
Comment (milmazz) [inline test/ex_doc/formatter/html/templates_test.exs]: You also need to check if there is a link present, right?

File: test/ex_doc/formatter/html/templates_test.exs [hunk 3]
```diff
@@ -83,19 +110,23 @@ defmodule ExDoc.Formatter.HTML.TemplatesTest do
...
   end
 
   test "listing page doesn't have README link if not present" do
     content = Templates.sidebar_template(doc_config, [], [], [], false)
-    refute content =~ ~r{<a href="README.html">README</a>}
+    links = Floki.find(content, ".sidebar-mainNav li")
+
+    assert_text Enum.at(links, 0), "Overview"
```
Comment (milmazz) [inline test/ex_doc/formatter/html/templates_test.exs]: What happens when someone change the order of the links in the menu? Imagine that first comes Overview and then README, this tests will pass but the README link will be present, right?

File: test/ex_doc/formatter/html/templates_test.exs (hunk 1 above)
Comment (Dignifiedquire) [inline test/ex_doc/formatter/html/templates_test.exs]: yes

**Comment by milmazz** [comment]
@Dignifiedquire Just out of curiosity, this is still pending, right?

**Comment by Dignifiedquire** [comment]
yes I'm still working on it

**Pull request closed without merging**